sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/CollapseRepartitionSuite.scala

```diff
@@ -32,6 +32,18 @@ class CollapseRepartitionSuite extends PlanTest {
 
   val testRelation = LocalRelation('a.int, 'b.int)
 
+
+  test("collapse two adjacent coalesces into one") {
+    val query = testRelation
+      .coalesce(10)
+      .coalesce(20)
```

Comment:
I think it would be better to respect the later input number, which is specified by users, for avoiding any surprise to users.